### PR TITLE
feat: Use argocd image for sidecar

### PR DESCRIPTION
**Pull request by jskrzypek**
### Description
This allows us to avoid having to manage the versions for helm & kustomize
alongside AVP and argocd, as these binaries are included in the argocd image.

We can add a kustomize images entry to specify / modify the image version from
the kustomization.yaml file.

**Fixes:** #368

### Checklist
Please make sure that your PR fulfills the following requirements:
- [x] Reviewed the guidelines for contributing to this repository
- [x] The commit message follows the [Conventional Commits Guidelines](https://www.conventionalcommits.org/en/v1.0.0/#summary).
- [ ] Tests for the changes have been updated
- [ ] Are you adding dependencies? If so, please run `go mod tidy -compat=1.17` to ensure only the minimum is pulled in.
- [ ] Docs have been added / updated
- [ ] Optional. My organization is added to USERS.md.

### Type of Change

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] New tests
- [ ] Build/CI related changes
- [ ] Documentation content changes
- [ ] Other (please describe)

### Other information
This is intended to be a draft PR just to illustrate my thought in #368. If the maintainers like it I can make it into an actual PR.

### CODE REVIEW

File: manifests/cmp-sidecar/argocd-repo-server.yaml [hunk 1]
```diff
@@ -24,28 +25,13 @@ spec:
         image: registry.access.redhat.com/ubi8
         env:
           - name: AVP_VERSION
-            value: 1.11.0
```
Comment (jkayani) [inline manifests/cmp-sidecar/argocd-repo-server.yaml:27]: We'll need to keep this line so that `AVP_VERSION` is defined

Comment (jskrzypek) [reply manifests/cmp-sidecar/argocd-repo-server.yaml:27]: Oops definitely not intended!

File: manifests/cmp-sidecar/argocd-repo-server.yaml [hunk 2]
```diff
@@ -24,28 +25,13 @@ spec:
...
-          - name: HELM_VERSION
-            value: 3.9.0
-          - name: KUSTOMIZE_VERSION
-            value: 4.5.5
         command: [sh, -c]
         args:
           - >-
             curl -L https://github.com/argoproj-labs/argocd-vault-plugin/releases/download/v$(AVP_VERSION)/argocd-vault-plugin_$(AVP_VERSION)_linux_amd64 -o argocd-vault-plugin &&
             chmod +x argocd-vault-plugin &&
             mv argocd-vault-plugin /custom-tools/ &&
```
Comment (jkayani) [inline manifests/cmp-sidecar/argocd-repo-server.yaml:33]: Need to remove this extra `&&` too

**Review by jkayani — COMMENTED** [review]
Barring the 2 nits pointed out, this worked when I tested it locally to deploy both a Helm chart and Kustomize app with AVP. I used this manifest:

```yaml
apiVersion: kustomize.config.k8s.io/v1beta1
kind: Kustomization

resources:
- github.com/jskrzypek/argocd-vault-plugin//manifests/cmp-sidecar?ref=jskrzypek/sidecar-from-argocd-image

patches:
- target:
    kind: Deployment
    name: argocd-repo-server
  patch: |-
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      name: argocd-repo-server
    spec:
      template:
        spec:
          containers:
            - name: avp
              env:
                - name: AVP_TYPE
                  value: vault
                - name: VAULT_ADDR
                  value: http://vault.default.svc:8200
                - name: AVP_AUTH_TYPE
                  value: k8s
                - name: AVP_K8S_ROLE
                  value: argocd
            - name: avp-helm
              env:
                - name: AVP_TYPE
                  value: vault
                - name: VAULT_ADDR
                  value: http://vault.default.svc:8200
                - name: AVP_AUTH_TYPE
                  value: k8s
                - name: AVP_K8S_ROLE
                  value: argocd
            - name: avp-kustomize
              env:
                - name: AVP_TYPE
                  value: vault
                - name: VAULT_ADDR
                  value: http://vault.default.svc:8200
                - name: AVP_AUTH_TYPE
                  value: k8s
                - name: AVP_K8S_ROLE
                  value: argocd
```

Address the comments and this looks good to me

**Comment by jskrzypek** [comment]
Nits fixed!

**Review by werne2j — APPROVED** [review]
Looks good to me!

File: manifests/cmp-sidecar/argocd-repo-server.yaml [hunk 3]
```diff
@@ -25,26 +26,12 @@ spec:
...
           - name: AVP_VERSION
             value: 1.11.0
-          - name: HELM_VERSION
-            value: 3.9.0
-          - name: KUSTOMIZE_VERSION
-            value: 4.5.5
         command: [sh, -c]
         args:
           - >-
             curl -L https://github.com/argoproj-labs/argocd-vault-plugin/releases/download/v$(AVP_VERSION)/argocd-vault-plugin_$(AVP_VERSION)_linux_amd64 -o argocd-vault-plugin &&
```
Comment (jskrzypek) [inline manifests/cmp-sidecar/argocd-repo-server.yaml:32]: FWIW, I realized yesterday (and didn't get to adding it to the PR) that this code could instead be 

```suggestion
curl -L https://github.com/argoproj-labs/argocd-vault-plugin/releases/download/v$(AVP_VERSION)/argocd-vault-plugin_$(AVP_VERSION)_linux_amd64 -o /custom-tools/argocd-vault-plugin &&
chmod +x /custom-tools/argocd-vault-plugin
```

YMMV, but the benefit here would be that the initContainer could be more secure and use 
```yaml
securityContext:
  readOnlyRootFilesystem: true
```
as well as not leaving the binary lying around on the host

File: manifests/cmp-sidecar/argocd-repo-server.yaml [hunk 4]
```diff
@@ -25,26 +26,12 @@ spec:
...
-            curl -L https://get.helm.sh/helm-v$(HELM_VERSION)-linux-amd64.tar.gz -o helm.tar.gz &&
-            tar -xvf helm.tar.gz &&
-            chmod +x linux-amd64/helm &&
-            mv linux-amd64/helm /custom-tools &&
-
-            curl -L https://github.com/kubernetes-sigs/kustomize/releases/download/kustomize%2Fv$(KUSTOMIZE_VERSION)/kustomize_v$(KUSTOMIZE_VERSION)_linux_amd64.tar.gz -o kustomize.tar.gz &&
-            tar -xvf kustomize.tar.gz &&
-            chmod +x kustomize &&
-            mv kustomize /custom-tools
+            mv argocd-vault-plugin /custom-tools/
```
Comment (jskrzypek) [inline manifests/cmp-sidecar/argocd-repo-server.yaml:34]: FWIW, I realized yesterday (and didn't get to adding it to the PR) that this code could instead be 

```suggestion
            curl -L https://github.com/argoproj-labs/argocd-vault-plugin/releases/download/v$(AVP_VERSION)/argocd-vault-plugin_$(AVP_VERSION)_linux_amd64 -o /custom-tools/argocd-vault-plugin &&
            chmod +x /custom-tools/argocd-vault-plugin
```

YMMV, but the benefit here would be that the initContainer could be more secure and use 
```yaml
securityContext:
  readOnlyRootFilesystem: true
```
as well as not leaving the binary lying around on the host

**Pull request merged**